src/main.c

```diff
@@ -353,8 +353,10 @@ int main(int argc, char* argv[]) {
   size_t short_opts = 0;
   jv lib_search_paths = jv_null();
   for (int i=1; i<argc; i++, short_opts = 0) {
-    if (args_done) {
-      if (further_args_are_strings) {
+    if (args_done || !isoptish(argv[i])) {
```

Comment:
Arguably we should stop parsing options the moment we see a non-option argument.

Reply:
No, we should not because that would be a very badly breaking change. D:
For scripts that use `jq --args '...' -- foo bar`, but also other applications.

Reply:
Yes, I know, but that's what I should have coded.  Though... it might be kinda useful to be able to alternate between `--args` and `--jsonargs`, as you noted?